processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -188,6 +188,22 @@ public BeanMappingMethod build() {
                             Message.BEANMAPPING_NOT_ASSIGNABLE, resultType, method.getResultType()
                         );
                     }
+                    else if ( resultType.isAbstract() ) {
+                        ctx.getMessager().printMessage(
+                            method.getExecutable(),
+                            beanMappingPrism.mirror,
+                            Message.BEANMAPPING_ABSTRACT,
+                            resultType,
+                            method.getResultType()
+                        );
+                    }
+                }
+                else if ( !method.isUpdateMethod() && method.getReturnType().isAbstract() ) {
```

Comment:
Are interfaces handled somewhere else already? I'd find it more clear if we'd also report an error (or the same error) if the given result type is an interface...